server/src/main/java/org/apache/iotdb/db/engine/compaction/cross/inplace/task/MergeMultiChunkTask.java

```diff
@@ -112,33 +115,42 @@ public MergeMultiChunkTask(
       InplaceCompactionLogger inplaceCompactionLogger,
       CrossSpaceMergeResource mergeResource,
       boolean fullMerge,
-      List<PartialPath> unmergedSeries,
+      List<PartialPath> unmergedDevice,
       int concurrentMergeSeriesNum,
       String storageGroupName) {
     this.mergeContext = context;
     this.taskName = taskName;
     this.inplaceCompactionLogger = inplaceCompactionLogger;
     this.resource = mergeResource;
     this.fullMerge = fullMerge;
-    this.unmergedSeries = unmergedSeries;
+    this.unmergedDevice = unmergedDevice;
     this.concurrentMergeSeriesNum = concurrentMergeSeriesNum;
     this.storageGroupName = storageGroupName;
   }
 
   void mergeSeries() throws IOException {
-    if (logger.isInfoEnabled()) {
-      logger.info("{} starts to merge {} series", taskName, unmergedSeries.size());
-    }
     long startTime = System.currentTimeMillis();
     for (TsFileResource seqFile : resource.getSeqFiles()) {
       // record the unmergeChunkStartTime for each sensor in each file
       mergeContext.getUnmergedChunkStartTimes().put(seqFile, new HashMap<>());
     }
+    mergedDeviceCnt = 0;
     // merge each series and write data into each seqFile's corresponding temp merge file
-    List<List<PartialPath>> devicePaths = MergeUtils.splitPathsByDevice(unmergedSeries);
-    for (List<PartialPath> pathList : devicePaths) {
+    for (PartialPath device : unmergedDevice) {
       // TODO: use statistics of queries to better rearrange series
-      IMergePathSelector pathSelector = new NaivePathSelector(pathList, concurrentMergeSeriesNum);
+      List<PartialPath> measurementPathListByDevice;
+      try {
+        measurementPathListByDevice =
+            new ArrayList<>(IoTDB.metaManager.getAllMeasurementByDevicePath(device));
```

Comment:
This methods, getAllMeasurementByDevicePath, returns measurementPath which contains schema. Maybe you could use the returned schema directly rather than query mmanager for second time.

Reply:
done